---
title: "pbta-tumor-evolution project - PBTA Cohort - Copy data file from: ./OpenPedCan-analysis/data/v12/histologies.tsv May 1st, 2023"
author: 'User ID: chronia'
date: "2023-05-05"
output:
  html_document:
    fig_height: 7
    fig_width: 9
    keep_md: yes
    toc: yes
    df_print: paged
params:
  out_path: "/Users/chronia/CHOP/projects/pbta-tumor-evolution/analysis/01_sample_distribution_analysis/"
  out_path_plots: "/Users/chronia/CHOP/projects/pbta-tumor-evolution/analysis/01_sample_distribution_analysis/plots/"
  out_path_results: "/Users/chronia/CHOP/projects/pbta-tumor-evolution/analysis/01_sample_distribution_analysis/results/"
  data_path: "/Users/chronia/CHOP/projects/pbta-tumor-evolution/analysis/01_sample_distribution_analysis/input/histologies.tsv" # path to dir with v12
  data_path_cols: "/Users/chronia/CHOP/projects/pbta-tumor-evolution/analysis/01_sample_distribution_analysis/input/PBTA-germline-histology-groups-plot_mapping.csv"
  sample_name: "PBTA"
  subsample: TRUE
---

#### Tumor evolution project 

# aim
This is an exploratory analysis of longitudinal data in the PBTA cohort.
We are investigating the number of histologies (cancer_type column): (1) per assay (experimental_strategy) and (2) pairs of genomic and transcriptomic assays.
We are also looking at the number of patient samples available per histology with paired genomic and transcriptomic assays for (1) each time point (disease_stage) and (2) when Diagnosis is present and paired with another point in time.

# data used 
In this notebook, we are looking into the v12 histologies file (last updates on May 1st, 2023) and subset to the PBTA cohort.
We use tumor_descriptor to summarize and filter disease_stages and generate "tumor_descriptor_sum", which wa sused for downstream analysis.

# note
The colors in the plots do not match the ones of the PBTA cohort yet. ---TO DO---



## Load libraries
```{r setup, include=FALSE}
attach(params) 
library(tidyverse)
library(writexl)
library(ggplot2)
library(magrittr) # needs to be run every time you start R and want to use %>%
library(ggthemes)
library(scales) # For pretty breaks
library(ggpubr)
library(broom)
library(viridis)
library(hrbrthemes)
library(stringi)
library(ggpubr)
library(rstatix)
library(dplyr)
library(RColorBrewer)
suppressPackageStartupMessages(library(purrr))
suppressPackageStartupMessages(library(furrr))

library(reshape2) 
library(reshape) 
library(flextable)

knitr::opts_chunk$set(echo = TRUE, 
                      warning = FALSE, 
                      message = FALSE, 
                      fig.path = file.path(out_path_plots),
                      dev = c("png"))

# enable parallelization
#plan("multiprocess", workers = 4) # 10 cpus
plan("multisession", workers = 4) # 10 cpus
options(future.globals.maxSize = 59600 * 1024^3) # for 64 Gb RAM, 119200 * 1024^3 for 128GB

# set out_path and create dir 
out_path <- if (!dir.exists(out_path)) {dir.create(out_path)}

# shuffle function:
shuf <- function(df){
  return(df[sample(1:dim(df)[1], dim(df)[1]),])
}

```

## Load data
```{r }
v12 <- readr::read_tsv(data_path)
# dim(data)
data_path_cols <- read.csv(data_path_cols, stringsAsFactor=FALSE) # Load plotting colors
# Add colors palette to the histologies file
hist_cols <- v12 %>%
left_join(cols, by=c('cancer_group','broad_histology')

# to discard when add the col palette for PBTA cohort
mypalette <- brewer.pal(12,"Paired") # colorblindFriendly = TRUE
col5 <- colorRampPalette(mypalette)(5)
col8 <- colorRampPalette(mypalette)(8)

```

All tumor samples have matched normal samples. 
We need to filter normal samples out for further analysis. 
We are using "pathology_diagnosis" instead of "sample_type". 
This way we discard normal samples (as shown in "sample_type") in addition to undiagnosed tumor samples.
From "composition" discard Derived Cell Line, PDX.

```{r cohort PBTA, echo=TRUE}
data <- data1  %>%
   filter(cohort == "PBTA",
          !is.na(pathology_diagnosis),
          !composition %in% c("Derived Cell Line", "PDX"),
)

data %>% count(tumor_descriptor)

# Save data after filter 
write_csv(data, file.path(out_path_results, out_file))

# Replicate tumor_descriptor column and rename to tumor_descriptor_sum. This will summarize some longitudinal info.
# data %>% count(tumor_descriptor)

data$tumor_descriptor_sum <- data$tumor_descriptor
data$tumor_descriptor_sum <- ifelse(data$tumor_descriptor_sum=="Progressive Disease Post-Mortem", "Deceased",
                                 ifelse(data$tumor_descriptor_sum=="Initial CNS Tumor", "Diagnosis",
                                 ifelse(data$tumor_descriptor_sum=="Primary Tumor", "Diagnosis", data$tumor_descriptor_sum)))
data <- data %>% filter(tumor_descriptor_sum %in% c("Diagnosis", "Progressive", "Recurrence", "Deceased")) 

data %>% count(tumor_descriptor_sum)

# Save data after filter 
write.csv(data, paste0(out_path_results, "data_by_cohort_", "PBTA", "_filter_logitudinal.csv"))

data %>% count(cancer_group)
data %>% count(tumor_descriptor)
data %>% count(tumor_descriptor_sum)
data %>% count(composition)
data %>% count(pathology_diagnosis)

count1 <- data %>% 
          group_by(sample_id, cohort_participant_id, Kids_First_Participant_ID,
                 experimental_strategy, sample_type, tumor_descriptor, tumor_descriptor_sum, primary_site,
                 age_at_diagnosis_days, age_last_update_days, age_at_chemo_start, age_at_radiation_start,
                 pathology_diagnosis, cancer_group, broad_histology) %>% 
          dplyr::count(cancer_group) 

write.csv(count1, paste0(out_path_results, "data_by_cohort_", "PBTA", "_filter_logitudinal_count.csv"))

```


## All assays

```{r plot1}
count1$experimental_strategy <- factor(x = count1$experimental_strategy, levels = c("WGS", "WXS", "RNA-Seq", "Targeted Sequencing", "Methylation"))

p <- ggplot(count1, aes(x=cancer_group, fill=experimental_strategy))+
        geom_bar()+
        scale_fill_manual(values=col5) +
        theme_classic()+
        ylab("Count cancer_group") +
        ggtitle("cancer_group vs experimental_strategy") +
        guides(color=guide_legend(override.aes=list(size=3))) +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
        print(p)
ggsave("Barplot_cancer_group_experimental_strategy_PBTA.pdf", plot=p, path=out_path_plots, device="pdf", height=7, width=10)

```

## Paired genomic, transcriptomic assays

```{r plot2}
count <- count1 %>% filter(experimental_strategy %in% c("WGS", "WXS", "RNA-Seq", "Targeted Sequencing")) # Not included:Methylation
count$experimental_strategy <- factor(x = count$experimental_strategy, levels = c("WGS", "WXS", "RNA-Seq", "Targeted Sequencing"))

p <- ggplot(count, aes(x=cancer_group, fill=experimental_strategy))+
        geom_bar()+
        scale_fill_manual(values=col5) + #scale_fill_manual(values=c(data$plot_group_hex)) 
        theme_classic()+
        ylab("Count cancer_group") +
        ggtitle("cancer_group vs experimental_strategy") +
        guides(color=guide_legend(override.aes=list(size=3))) +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
        print(p)
ggsave(paste0("Barplot_cancer_group_experimental_strategy_matched_DNA-RNA_", "PBTA", ".pdf"), plot=p, path=out_path_plots, device="pdf", height=7, width=10)
```

## By cancer_group - all disease stages


```{r plot3}
count$tumor_descriptor_sum <- factor(x = count$tumor_descriptor_sum, levels = c("Diagnosis", "Progressive", "Recurrence", "Deceased"))

p <- ggplot(count, aes(x=cancer_group, fill=tumor_descriptor_sum))+
        geom_bar()+
        scale_fill_manual(values=col5) +
        theme_classic()+
        ylab("Count cancer_group") +
        ggtitle("cancer_group vs tumor_descriptor_sum") +
        guides(color=guide_legend(override.aes=list(size=3))) +
        theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
        print(p)
ggsave(paste0("Barplot_cancer_group_tumor_descriptor_sum_matched_DNA-RNA_", "PBTA", ".pdf"), plot=p, path=out_path_plots, device="pdf", height=7, width=10)

```

In the following section, we will provide summary table across different disease stages, histologies, and patients samples.

### Number of tumor samples per each assay

First, weâ€™ll examine how many of each type of assay we have (tumors only). This information is stored in the experimental_strategy column.

```{r}
count1 %>%
  group_by(experimental_strategy) %>%
  tally() %>%
  arrange(desc(n)) %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")

```



### Paired genomic, transcriptomic assays

## Multiple time points: Diagnosis, Progressive, Recurrence, Deceased
We're going to use the `tumor_descriptor_sum` column here with matched genomic and transcriptomic assays from before.

## By cancer_group 
# Disease stage - descriptors pairs

How many tumor samples per cancer_group and disease_stage?

```{r}
disease_stage_cancer_group <- count %>%
  group_by(Kids_First_Participant_ID) %>%
  summarize(disease_stage = paste(sort(unique(tumor_descriptor_sum)),
                                  collapse = ", "),
            cancer_group = paste(sort(unique(cancer_group)),
                                collapse = ", ")) %>%
  group_by(disease_stage, cancer_group) %>%
  tally() %>%
  arrange(desc(n))

disease_stage_cancer_group %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")

```


What about when Diagnosis is present and paired with another point in time?

```{r}
Diagnosis_paired <- disease_stage_cancer_group  %>%
  filter(disease_stage != "Diagnosis")  %>% 
  filter(grepl("Diagnosis", disease_stage)) %>%
  regulartable() %>%
  fontsize(size = 12, part = "all")
Diagnosis_paired
```



## By Kids_First_Participant_ID
### Disease stage - descriptors pairs

How many tumor samples per disease_stage?

```{r}
disease_stage <- count %>%
  group_by(Kids_First_Participant_ID) %>%
  summarize(disease_stage = paste(sort(unique(tumor_descriptor_sum)),
                                  collapse = ", "),
                                collapse = ", ") %>%
  group_by(disease_stage) %>%
  tally() %>%
  arrange(desc(n))

disease_stage  %>% 
  regulartable() %>%
  fontsize(size = 12, part = "all")

```


What about when Diagnosis is present and paired with another point in time?

```{r}
Diagnosis_paired <- disease_stage  %>%
  filter(disease_stage != "Diagnosis")  %>% 
  filter(grepl("Diagnosis", disease_stage)) %>%
  regulartable() %>%
  fontsize(size = 12, part = "all")
Diagnosis_paired
```


